crates/ruff/resources/test/fixtures/flake8_bugbear/B006_B008.py

```diff
@@ -67,6 +67,11 @@ def this_is_wrong(value=[1, 2, 3]):
 def this_is_also_wrong(value={}):
     ...
 
+class Foo:
+    @staticmethod
+    def this_is_also_wrong_and_more_indented(value={}):
+        pass
+
```

Comment:
another test case i'd like to see is a multi-line default value

Reply:
We also need to handle:

```python
def f(value = {}): ...
```

Reply:
So multiline arg seems to do fine, but same-line func creates defunct Python. It's best to not fix these indeed. Is there a quick way to check if a function is one line, or its body is on the same line as its def?

Reply:
`locator.contains_line_break` between the colon (`:`) and the first body statement should work

Reply:
(to avoid misunderstandings: i believe multi line args are fixable, they should just get a test like the one you added. Same line functions on the other hand would require manual indentation handling and i don't think it's worth implementing this)

Reply:
Added a check for single-line func and changed from `AlwaysAutoFixableViolation` to `Violation` and sometimes fixable (also already changed the fixes to manual as per request in another comment so unsure what type of Violation the rule should now be to be fair, as so far it appears to be the only rule using `manual_edit(s)`)

Reply:
Hmm my heuristic is not covering all cases in the end.

```rust
checker
                    .locator
                    .contains_line_break(TextRange::new(arguments.range().end(), body[0].start()))
```

Allows weird things like 

```python
def single_line_func_wrong(value = {}

                           )\
    : ...
```

To slip through. Is there a way to efficiently locate the range value of the `:`? I have found the `lexer::lex` but that would require getting a string representation of the function def. Is that the way to go, or is there a better way to locate single characters in parts of a `StmtFunctionDef`? Maybe a `find_x_in_range` like util or something along those lines?

Reply:
Using the lexer when creating a diagnostic can be useful, but it is preferred not to when possible. But I don't see how you can get away with not using the lexer here (we wrote a more lightweight lexer in the formatter, because getting to tokens is such a common task, maybe it's time to make it available to ruff too). You can get the string representation of the function def by using `locator.slice(def.range())`. You can then start lexing *after* the last argument, find the `colon`. Then use `locator.contains_line_break(TextRange::new(colon.end(), first_statement.start()`

Which raises an interesting question. Is this valid? Is this a suite or a single statement body 

```python
def single_line_func_wrong(value = {}):\
	...
```

Unrelated: @konstin a use case where `StmtSuite` would be handy... It would tell us right away whether it is a single statement or a suite.

Reply:
Thanks for the hyperclear instructions and context. I'll see if I can spot and autofix the 
```python
def single_line_func_wrong(value = {}):\
    ...
```
scenario without too much hassle and, if so, include it and only leave out the 
```python
def single_line_func_wrong(value = {}): ...
```
scenario.

Reply:
So I've managed to exclude `def single_line_func_wrong(value = {}): ...` based on your instructions. The other scenario you raise is tricky though. The lexer does not do anything with the `\` and it counts as a line break for `contains_line_break()` so current implementation still tries to make:
```python
def single_line_func_wrong(value = None):\
    if value is None:
        value = {}
```
of it, which is broken. Is there a way to discern `\` line breaks from other line breaks? Or do we want to go a different route here?